src/components/DropdownMenu.tsx

```diff
@@ -0,0 +1,99 @@
+
+import { ArchiveBox } from 'phosphor-react'
+import { EditorContentContext } from '../contexts/EditorContentContext'
+
+import JSZip from 'jszip'
+import { saveAs } from 'file-saver'
+import pretty from 'pretty'
+
+const zip = new JSZip()
+
+export function DropdownMenu() {
+  const { app } = useContext(EditorContentContext)
+
+  function addScriptsToParsedHtmlHeader(parsed: any) {
+    const header = parsed.querySelector('head')
+
+    const script = document.createElement('script')
+    script.type = 'text/javascript'
+    script.src = './index.js'
+    script.defer = true
+
+    const style = document.createElement('link')
+    style.rel = 'stylesheet'
+    style.href = './index.css'
+
+    header.appendChild(script)
+    header.appendChild(style)
+
+    return parsed
+  }
+
+  function handleDownloadAsZip() {
+    const parser = new DOMParser()
+
+    const parsedHTML = parser.parseFromString(app.html, 'text/html')
+
+    const htmlWithScripts = addScriptsToParsedHtmlHeader(parsedHTML)
+
+    zip.file(
+      'index.html',
+      pretty(new XMLSerializer().serializeToString(htmlWithScripts)),
+    )
+
+    zip.file('index.css', app.css)
+    zip.file('index.js', app.javascript)
+    zip.file('index.md', app.markdown)
+
+    zip.generateAsync({ type: 'blob' }).then(function (content) {
+      saveAs(
+        content,
+        `frontend-editor-${Math.random().toString(36).slice(2)}.zip`,
+      )
+    })
+  }
+
+  return (
+    <div className="top-16">
+      <Menu as="div" className="relative inline-block text-left">
+        <div>
+          <Menu.Button className="inline-flex items-center w-full justify-center rounded-md bg-black bg-opacity-20 px-4 py-2 text-sm font-medium text-white hover:bg-opacity-30 focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75">
+            <img src={logoSvg} className="w-4 inline mr-2" alt="" />
+            Options
+          </Menu.Button>
+        </div>
+        <Transition
+          as={Fragment}
+          enter="transition ease-out duration-100"
+          enterFrom="transform opacity-0 scale-95"
+          enterTo="transform opacity-100 scale-100"
+          leave="transition ease-in duration-75"
+          leaveFrom="transform opacity-100 scale-100"
+          leaveTo="transform opacity-0 scale-95"
+        >
+          <Menu.Items className="absolute mt-2 w-56 origin-top-right divide-y divide-gray-100 rounded-md bg-white shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none">
+            <div className="px-1 py-1 ">
+              <Menu.Item>
+                {({ active }) => (
+                  <button
+                    className={`${
+                      active ? 'bg-green-400 text-gray-900' : 'text-gray-900'
+                    } group flex w-full items-center rounded-md px-2 py-2 text-sm`}
+                    onClick={handleDownloadAsZip}
+                  >
+                    <ArchiveBox size={21} className="mr-2" />
```

Comment:
21 give me goosebumps, can we achieve 20 or 24?

Reply:
Instead of `mr-2` I always prefer a `gap-2` in the container, what do you think?